### ISSUE TITLE
editColumn doesn't work for column sorting/ordering?

### ISSUE THREAD

**Issue by intrepidws** [yajra/laravel-datatables#535]
### Summary of problem or feature request

I'm using editColumn() (per http://datatables.yajrabox.com/eloquent/carbon) to show the time portion of a date in a separate field. However, when sorting/ordering by this field, instead of using the results from editColumn(), the entire date field is used.

I tried using orderColumn() (per http://datatables.yajrabox.com/eloquent/order-column) but the documentation is confusing and I'm not even sure if that's what it should be used for. It seems only applicable to order one column instead by another column?

### Code snippet of problem

```php
->editColumn('time', function ($event) {
    return date('h:i', strtotime($event->date)) . substr(date('a', strtotime($event->date)), 0, 1);
})
```

### System details

- Operating System: Ubuntu 14.04
- PHP Version: PHP 5.6.10
- Laravel Version: 5.0.35
- Laravel-Datatables Version: 6.10.1

**Comment by yajra** [comment]
This is a limitation on the editColumn function since we are sorting on the database and not on the returned collection. A workaround would be to use `orderColumn`. orderColumn uses `orderByRaw` query with `$1` variable as order direction. Using this method, you can write the order the way you display it using editColumn.

>It seems only applicable to order one column instead by another column?

It's just for demo purposes that I literally change the order of name to email. As stated above, you can write your own raw query on `orderColumn` which uses `orderByRaw` internally.

**Comment by intrepidws** [comment]
Is there a difference between `orderColumn` and `order`?

**Comment by intrepidws** [comment]
For anyone that this might help, I ended up using the following:

```php
orderColumn('events.date', 'DATE_FORMAT(events.date, "%y%m%d") $1')
```

**Comment by AnubisTheory** [comment]
Is it possible to use a closure instead of raw SQL instead? Or at least have access the to the columns (as variables) the way you do when doing editColumn()?

**Comment by cbloss** [comment]
I don't know if this helps anyone, but I had a situation where I needed to sort on a column that I modified using editColumn and still needed the original column. For example, I had this in a column called path that got returned from the database:

aaa/bbb/ccccccc

The last little bit (cccccc) could also have another / in there. With the `editColumn`, I returned the lastlittlbit with additional / if it was in there. By default, the Datatables ordered the original column so with the aaa, but not the cccccc. Using `orderColumn`, I was able to figure it out. Here's what I came up with: 

`->orderColumn('path', 'SUBSTRING(path FROM (LOCATE("/",path)+LOCATE("/", path) + 1))')`

This forced DataTables to order the path column based on the substring and not what was returned in the database. 

Hope this helps someone who came across this!
